third-party/realdds/include/realdds/dds-option.h

```diff
@@ -20,37 +22,38 @@ struct dds_option_range
     float default_value = 0.0f;
 };
 
+class dds_stream_base;
+
 class dds_option
 {
     std::string _name;
     float _value = 0.0f;
     dds_option_range _range;
     std::string _description;
 
-    std::string _owner_name; //To identify device or stream having this option
+    dds_option( nlohmann::json const & );
 
-    dds_option( nlohmann::json const & j, const std::string & owner_name );
+    friend class dds_stream_base;
+    std::weak_ptr< dds_stream_base > _stream;
+    void init_stream( std::shared_ptr< dds_stream_base > const & );
```

Comment:
I don't like making this private and `dds_stream_base` as a friend. Why limit only stream to set itself in the option? Maybe we will want some factory or other entity that will handle the matching.

Reply:
We have a similar mechanism in `dds_stream_profile`, though there we did not use a friend. White I generally understand your aversion, I really do think that's the proper way to do it: streams and options are closely related and it's only the stream base that's a friend. That's how we currently associate streams and options: through the `init_options()` call.

As to "Maybe we will want some factory or other entity that will handle the matching", unless you have such a mechanism in mind, the current implementation is sufficient...

If you insist, I can unfriend it and make it public, but I prefer leaving like this.